syntax = "proto3";

package vault;

service VaultService {
    rpc HealthService (HealthRequest) returns (HealthResponse);
    rpc SearchService (SearchRequest) returns (SearchResponse);
    rpc IndexService (IndexRequest) returns (IndexResponse);
}

message HealthRequest {
}

message HealthResponse {
    string status = 1;
}

message SearchRequest {
    string query = 1;
    int32 k = 2;
}

message SearchResult {
    string content = 1;
    string path = 2;
    float score = 3;
}

message SearchResponse {
  repeated SearchResult results = 1;
}


message IndexRequest {
  string path = 1;
}

message IndexResponse {
  string status = 1;
}


